### PR TITLE
Add fail_on_exception flag

### PR DESCRIPTION
**Pull request by HubertJaworski**

### CODE REVIEW

File: neptune/new/internal/init_impl.py [hunk 1]
```diff
@@ -326,7 +329,8 @@ def init(project: Optional[str] = None,
             run_id=api_run.short_id
         ))
 
-    uncaught_exception_handler.activate()
+    if fail_on_exception:
```
Comment (aniezurawski) [inline neptune/new/internal/init_impl.py:332]: Is this `if` necessary? In future we can have many handlers registered.

File: neptune/new/internal/utils/traceback_job.py [hunk 2]
```diff
@@ -31,18 +31,21 @@
...
+    def __init__(self, path: str, fail_on_exception: bool = True):
         self._uuid = uuid.uuid4()
         self._started = False
         self._path = path
+        self._fail_on_exception = fail_on_exception
 
     def start(self, run: 'Run'):
         if not self._started:
             path = self._path
+            fail_on_exception = self._fail_on_exception
```
Comment (aniezurawski) [inline neptune/new/internal/utils/traceback_job.py:43]: What is it for?

**Review by aniezurawski — CHANGES_REQUESTED** [review]

File: neptune/new/internal/init_impl.py (hunk 1 above)
Comment (HubertJaworski) [reply neptune/new/internal/init_impl.py:332]: No, I should remobe this

File: neptune/new/internal/utils/traceback_job.py (hunk 2 above)
Comment (HubertJaworski) [reply neptune/new/internal/utils/traceback_job.py:43]: because you can't reference self easilsy from a `def` function passed as lambda

**Review by aniezurawski — APPROVED** [review]

**Pull request merged**